tests/integration/models/torch/lighting_modules/test_segmentation.py

```diff
@@ -0,0 +1,163 @@
+    def __init__(self, num_samples: int, batch_size: int = 4):
+        super().__init__()
+        self.num_samples = num_samples
+        self.batch_size = batch_size
+
+    def setup(self, stage= None):
+        self.dataset = DummyDataset(self.num_samples)
+
+    def train_dataloader(self):
+        return DataLoader(self.dataset, batch_size=self.batch_size)
+
+    def val_dataloader(self):
+        return DataLoader(self.dataset, batch_size=self.batch_size)
+
+    def test_dataloader(self):
+        return DataLoader(self.dataset, batch_size=self.batch_size)
+
+@pytest.fixture
+def segmentation_module() -> LightningModule:
+    cfg = DictConfig(
+        {
+            "models": fixt_models.deeplabv3_plus_w_target,
+            "trainer": fixt_trainers.trainer_cfg_w_cpu_devices,
+            "losses" : fixt_losses.jaccard_loss_w_target
+        }
+    )
+    model = get_model(cfg.models, cfg.trainer)
+    losses = get_losses(cfg, "image-segmentation", Frameworks.torch) #fixt_losses.jaccard_loss_w_target
+    optimizer_cfg = DictConfig(fixt_optimizers.adam_optim_w_target)
+
+    module = SemanticSegmentationLightningModule(
+        model=model, losses=losses, optimizer_cfg=optimizer_cfg
+    )
+
+    return module
+
+
+## working ## 
+
+def test_training_with_checkpoint(segmentation_module: LightningModule):
+    # Create a directory to save checkpoints
+    checkpoint_dir = "checkpoints"
+    os.makedirs(checkpoint_dir, exist_ok=True)
+
+    # Set up the ModelCheckpoint callback
+    checkpoint_callback = ModelCheckpoint(
+        dirpath=checkpoint_dir,
+        filename="segmentation_module-{epoch:02d}",
+        save_top_k=1,  # Save only the top 1 checkpoints
+        monitor="loss",  # You may need to change this to the appropriate metric you want to monitor
+        mode="min",  # Minimize the loss
+        save_weights_only=True,
+    )
+
+    # Training with checkpoint
+    trainer = Trainer(max_epochs=1, devices=1, callbacks=[checkpoint_callback])
+
+    # Create the dummy dataset and dataloader
+    dataset = DummyDataset(num_samples=100)
+    dataloader = DataLoader(dataset, batch_size=4)
+
+    trainer.fit(segmentation_module, train_dataloaders=dataloader)
+
+## working ##
+
+def test_training_without_checkpoint(segmentation_module: LightningModule):
+    # training without checkpoint
+    trainer = Trainer(max_epochs=1, devices=1)
+
+    # Create the dummy dataset and dataloader
+    dataset = DummyDataset(num_samples=100)
+    dataloader = DataLoader(dataset, batch_size=4)
+
+    trainer.fit(segmentation_module, train_dataloaders=dataloader)
+
+
+# working ##
+def test_testing_without_checkpoint(segmentation_module: LightningModule):
+    # Set up the trainer
+    trainer = Trainer(max_epochs=1, devices=1)
+
+    dataset = DummyDataset(num_samples=100)
+    dataloader = DataLoader(dataset, batch_size=4)
+
+    # Fit the model
+    trainer.test(segmentation_module, dataloaders=dataloader)
+
+
+
+
+
```

Comment:
Are you using pre-commit? let's delete this commented code if it's not needed

Reply:
thanks for noting the style problem. I will be fixing it

Reply:
https://innofw.readthedocs.io/en/latest/contributing/ step 4.